dbms/src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -1689,13 +1730,18 @@ void InterpreterSelectQuery::executeFetchColumns(
                 if (processing_stage == QueryProcessingStage::Complete)
                     source->addTotalsPort();
 
-                if (pin_sources)
-                    source->setStream(sources.size());
+                pipes.emplace_back(std::move(source));
+            }
 
-                sources.emplace_back(std::move(source));
+            /// Pin sources for merge tree tables.
+            bool pin_sources = dynamic_cast<const MergeTreeData *>(storage.get()) != nullptr;
```

Comment:
Is this code also temporary, because of MergeTree the first storage which supports processors?

Reply:
No.
I actually want to delete this code later.